lib/src/interfaces/a_star.dart

```diff
@@ -54,13 +58,88 @@ class AutonomyAStarState extends AStarState<AutonomyAStarState> {
     orientation: orientation, 
     direction: direction,
     goal: goal, 
-    depth: direction == DriveDirection.forward ? depth + 1 : depth + 2,
+    depth: (direction == DriveDirection.forward)
+        ? depth + 1
+        : (direction == DriveDirection.forwardLeft || direction == DriveDirection.forwardRight)
+            ? depth + sqrt2
+            : depth + 2,
   );
 
+  bool drivingThroughObstacle(AutonomyAStarState state) {
+    final isTurn = state.direction != DriveDirection.forward;
+    final isQuarterTurn = state.direction == DriveDirection.forwardLeft || state.direction == DriveDirection.forwardRight;
+
+    // Forward drive across the perpendicular axis
+    if (!isTurn && state.orientation.angle.abs() % 90 == 0) {
```

Comment:
Can you give `state.orientation.angle.abs() % 90 == 0` a descriptive variable name?